mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -812,18 +812,18 @@ func resourceContainerCluster() *schema.Resource {
 							Type:        schema.TypeList,
 <% if version == "ga" -%>
 							Required:    true,
-							Description: `GKE components exposing metrics. Valid values include SYSTEM_COMPONENTS.`,
+							Description: `GKE components exposing metrics. Valid values include SYSTEM_COMPONENTS, APISERVER, CONTROLLER_MANAGER, and SCHEDULER.`,
 <% else -%>
 							Optional:    true,
 							Computed:    true,
-							Description: `GKE components exposing metrics. Valid values include SYSTEM_COMPONENTS and WORKLOADS.`,
+							Description: `GKE components exposing metrics. Valid values include SYSTEM_COMPONENTS, APISERVER, CONTROLLER_MANAGER, SCHEDULER, and WORKLOADS.`,
 <% end -%>
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 <% if version == "ga" -%>
-								ValidateFunc: validation.StringInSlice([]string{"SYSTEM_COMPONENTS"}, false),
+								ValidateFunc: validation.StringInSlice([]string{"SYSTEM_COMPONENTS", "APISERVER", "CONTROLLER_MANAGER", "SCHEDULER"}, false),
```

Comment:
Is the number of values here expected to grow? In general, if an enum is adding values more than once a year it's best to drop it to a string to give users more flexibility with provider versions. We lose "plan-time" errors, as validation of the value drops to the API (making the error "apply-time") in doing so, which is a negative, but playing catch-up with changing enums and forcing provider version upgrades typically hurts users more.

Reply:
Thanks for the insight! This is a very good point!

Yes, the number of values may grow over time. There is no planned change on our radar at this moment. I think we may keep the validation for now, and drop it when we believe the frequency of adding values is more than once a year. WDYT?

Reply:
SGTM!